File: app/src/main/java/cenergy/central/com/pwb_store/adapter/viewholder/SearchProductViewHolder.java
package cenergy.central.com.pwb_store.adapter.viewholder;

import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.ImageView;
import android.widget.TextView;

import org.greenrobot.eventbus.EventBus;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnTextChanged;
import cenergy.central.com.pwb_store.R;
import cenergy.central.com.pwb_store.manager.bus.event.BarcodeBus;
import cenergy.central.com.pwb_store.manager.bus.event.SearchEventBus;
import cenergy.central.com.pwb_store.view.PowerBuyEditText;

/**
 * Created by napabhat on 7/11/2017 AD.
 */

public class SearchProductViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnFocusChangeListener{
    private static final String TAG = SearchProductViewHolder.class.getSimpleName();

    @BindView(R.id.edit_text_search)
    PowerBuyEditText mSearchView;

    @BindView(R.id.image_view_barcode)
    ImageView mBarCode;

    private boolean isSearch;

    public SearchProductViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        mBarCode.setOnClickListener(this);
        //mSearchView.setOnFocusChangeListener(this);
        mSearchView.setOnEditorActionListener(new SearchOnEditorActionListener());
    }

    public void setViewHolder(){

    }

    @Override
    public void onClick(View v) {
        if (v == mBarCode){
            EventBus.getDefault().post(new BarcodeBus(true));
            Log.d(TAG , " Click");
        }
    }

    @Override
    public void onFocusChange(View v, boolean hasFocus) {
        if (v == mSearchView){
            //EventBus.getDefault().post(new SearchEventBus(v, true));
            Log.d(TAG, "Search");
        }
    }

//    @OnTextChanged(R.id.edit_text_search)
//    public void textMoneyChanged (CharSequence text) {
//            if (text.length() > 0){
//                EventBus.getDefault().post(new SearchEventBus(true, mSearchView.getText().toString()));
//                Log.d(TAG, "Search");
//            }
//    }

    @OnTextChanged(value = R.id.edit_text_search,
            callback = OnTextChanged.Callback.TEXT_CHANGED)
    protected void onSearchChanged(CharSequence s, int start, int before, int count) {
        if (s.length() > 0) {
           // EventBus.getDefault().post(new SearchEventBus(true, mSearchView.getText().toString()));
        }
    }

    private class SearchOnEditorActionListener implements TextView.OnEditorActionListener {
        @Override
        public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
            if (actionId == EditorInfo.IME_ACTION_SEARCH || actionId == EditorInfo.IME_ACTION_UNSPECIFIED) {
                //performSearch();

                EventBus.getDefault().post(new SearchEventBus(true, mSearchView.getText().toString()));
            }

            return true;
        }
    }
}

File: app/src/main/java/cenergy/central/com/pwb_store/adapter/viewholder/SearchProductViewHolder.kt
package cenergy.central.com.pwb_store.adapter.viewholder

import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import cenergy.central.com.pwb_store.BuildConfig
import cenergy.central.com.pwb_store.R
import cenergy.central.com.pwb_store.manager.bus.event.BarcodeBus
import cenergy.central.com.pwb_store.manager.bus.event.HomeBus
import cenergy.central.com.pwb_store.manager.bus.event.SearchEventBus
import cenergy.central.com.pwb_store.view.PowerBuyEditText
import org.greenrobot.eventbus.EventBus

class SearchProductViewHolder(itemView: View, private val canBack: Boolean)
    : RecyclerView.ViewHolder(itemView), View.OnClickListener {

    private val mSearchView: PowerBuyEditText = itemView.findViewById(R.id.edit_text_search)
    private val mBarCode: ImageView = itemView.findViewById(R.id.image_view_barcode)
    private val backButton: ImageView = itemView.findViewById(R.id.image_back_button)

    override fun onClick(v: View) {
        when (v) {
            mBarCode -> {
                EventBus.getDefault().post(BarcodeBus(true))
            }
            backButton -> {
                EventBus.getDefault().post(HomeBus(true))
            }
        }
    }

    fun bind() {
        if (canBack){
            backButton.visibility = View.VISIBLE
            backButton.setOnClickListener(this)
        } else {
            backButton.visibility = View.GONE
            backButton.setOnClickListener(null)
        }
        if (BuildConfig.FLAVOR !== "pwbOmniTv") {
            mBarCode.setOnClickListener(this)
        } else {
            mBarCode.setOnClickListener(null)
            mBarCode.visibility = View.GONE
        }
        mSearchView.hint = itemView.context.getString(R.string.search_hint)
        mSearchView.setOnEditorActionListener(SearchOnEditorActionListener())
    }

    private inner class SearchOnEditorActionListener : TextView.OnEditorActionListener {
        override fun onEditorAction(v: TextView, actionId: Int, event: KeyEvent?): Boolean {
            if (actionId == EditorInfo.IME_ACTION_SEARCH || actionId == EditorInfo.IME_ACTION_UNSPECIFIED) {
                EventBus.getDefault().post(SearchEventBus(true, mSearchView.text.toString()))
            }
            return true
        }
    }
}
